### PR TITLE
Replace torch_scatter by native torch scatter ops

### PR DESCRIPTION
**Pull request by Kh4L**
Drop `torch_scatter` dependency 

Benchmark acc and speed:
| Method           | Time      | Avg Loss | Eval MRR |
|-----------------|----------|---------|---------|
| torch_scatter  | 4m31.629s | 3.7336  | 0.4884  |
| native scatter | 4m30.713s | 3.72461 | 0.4958  |

No statistically significant metric or speed difference between torch_scatter and native torch  

Also unused code with torch_scatter calls

### CODE REVIEW

**Comment by RManLuo** [comment]
> I reviewed the codes, and they seem good to me. The experiment performance with the released 8M model is the same as the previous implementation.
> 
> My only concern is the eps in native_scatter_softmax, which is not consistent with the official softmax implementation. It could lead to some precision differences during training.

It seems to be aligned with the implementation in`torch_spase`: https://pytorch-scatter.readthedocs.io/en/1.4.0/_modules/torch_scatter/composite/softmax.html

I have no more concerns now. Thanks again for your wonderful contribution!
